### DISCUSSION TITLE
Can't use Scanner (no HTTPS)

### DISCUSSION THREAD

**Discussion by TjardF** [Donkie/Spoolman#110] [General]
Hi,
I installed the container but it doesn't support https, is there something I have to change?
Maybe I have to create certificates on my own?
I'd like contribute some documentation on usage, integration and install later....
![image](https://github.com/Donkie/Spoolman/assets/37928929/8186a499-22d5-4207-88e5-f7ca2f3f9f3f)
![image](https://github.com/Donkie/Spoolman/assets/37928929/7bf270bc-312e-47a5-9be6-89de69a8adfd)

**Comment by Donkie** [comment]
Hi, yes exactly. You would need to setup a proxy server that can handle a certificate, nginx for example. There are many guides online on how to do this.

Here is just one tutorial I found that seems to fit the bill. https://seanthegeek.net/1035/how-to-configure-a-nginx-reverse-proxy-with-lets-encrypt-certificates/
The upstream site they are talking about is Spoolman then in this case.

**Reply by deadly667** [reply]
Hi,
first of all thanks for great app. Only one comment regarding this from my side. Is it maybe better to put flag as optional override for that if user wants (that scanner is also available on http). 

I personally don't want to expose my instance to public internet. On local home network, it's really overhead to have https and reverse proxy just for scanning QR code. I can always do that with custom QR code which will directly lead to my local instance of Spoolman but it's not point in that. I think that forcing https is maybe too much.

**Reply by Disane87** [reply]
> I think that forcing https is maybe too much

Thats not a spoolman constraint, it's a browser constraint:
https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia

And thats a pretty good contraint security wise.

**Comment by fleXXo22** [comment]
I found a way to disable the requirement for HTTPS camera access for specific websites.
Here's a guide: https://medium.com/@Carmichaelize/enabling-the-microphone-camera-in-chrome-for-local-unsecure-origins-9c90c3149339

It works for me.

**Reply by Disane87** [reply]
I would not recommend this because this is potentially unsafe.

If a website without SSL is requesting your camera and you have permission (there are ways to enable the permission without any consent but hacky hacky) your stream will be sent over the internet unencrypted and visible for everyone

**Comment by drlauridsen** [comment]
> I would not recommend this because this is potentially unsafe.
> 
> If a website without SSL is requesting your camera and you have permission (there are ways to enable the permission without any consent but hacky hacky) your stream will be sent over the internet unencrypted and visible for everyone

But in chrome, with `chrome://flags/#unsafely-treat-insecure-origin-as-secure`, you simply add only the local spoolman ip:port to the exception - this should not pose a risk for other ip/websites?

But since the feature of using the camera to scan QR  codes is a feature of spoolman, would it be possible to add https with a local certificate to the webserver? Or does camera access not accept local certificates?

**Reply by lfpoulain** [reply]
I think it shouldn't be that hard, there is this from the uvicorn server documentation : https://www.uvicorn.org/deployment/#running-with-https

**Comment by beikeland** [comment]
I know this is an old issue, maybe this has changed but currently if you access the site from `localhost` then you do not need HTTPS to use the camera. So even if I run spoolman for testing on a Pi on my local network I just make a tunnel from my desktop `ssh -L 7912:localhost:5173 pi@spoolman.local` and browser to http://localhost:7912/ with no restrictions.

(My "prod" Spoolman is accessed over public HTTPS using Cloudflared with their zero trust tunnel and access policies)
